main.js

```diff
@@ -1,18 +1,40 @@
 import messenger from './src/js/postMessenger';
 
+const sendMonitoringEvent = message =>
+	messenger.post({ type: 'oAds.monitor', message }, window.top);
+
 const handleReceivedMessage = event => {
-	if (event.origin === 'https://ft.com' && event.data) {
-		if (event.data.messageType === 'oAdsEmbed') {
+	if (event.data && event.data.messageType === 'oAdsEmbed') {
+		if (event.origin === 'https://ft.com') {
```

Comment:
I am not familiar with this codebase. Is this supposed to work on any app that uses oAds? Does it work on the ft-app?

Reply:
Not yet. This is something that only works on ft.com. I am sending this message from n-ads: https://github.com/Financial-Times/n-ads/blob/625223f203d4811e2f0b11653a86f6b7b233f1e9/src/client/smartmatch.js#L9